php_phongo.c

```diff
@@ -1275,55 +1288,71 @@ void php_phongo_objectid_new_from_oid(zval *object, const bson_oid_t *oid TSRMLS
 	bson_oid_to_string(oid, intern->oid);
 } /* }}} */
 
-void php_phongo_server_to_zval(zval *retval, const mongoc_server_description_t *sd) /* {{{ */
+php_phongo_server_description_type_t php_phongo_server_description_type(mongoc_server_description_t *sd)
 {
-	array_init(retval);
-
-	ADD_ASSOC_STRING(retval, "host", (char *)sd->host.host);
-	ADD_ASSOC_LONG_EX(retval, "port", sd->host.port);
-	ADD_ASSOC_LONG_EX(retval, "type", sd->type);
-	ADD_ASSOC_BOOL_EX(retval, "is_primary", sd->type == MONGOC_SERVER_RS_PRIMARY);
-	ADD_ASSOC_BOOL_EX(retval, "is_secondary", sd->type == MONGOC_SERVER_RS_SECONDARY);
-	ADD_ASSOC_BOOL_EX(retval, "is_arbiter", sd->type == MONGOC_SERVER_RS_ARBITER);
-	{
-		bson_iter_t iter;
-		zend_bool b = bson_iter_init_find_case(&iter, &sd->last_is_master, "hidden") && bson_iter_as_bool(&iter);
+	const char* name = mongoc_server_description_type(sd);
+	int i;
 
-		ADD_ASSOC_BOOL_EX(retval, "is_hidden", b);
+	for (i = 0; i < PHONGO_SERVER_DESCRIPTION_TYPES; i++) {
+		if (!strcmp(name, php_phongo_server_description_type_map[i].name)) {
```

Comment:
How often, and when is this run? A whole loop is going to be slow. If this runs a lot, we should re-order the items in the map, so that the most likely show up first.